### PR TITLE
Add `isEligibleToContributeAndFinalize` method to `Task`

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/core/task/Task.java [hunk 1]
```diff
@@ -178,4 +178,19 @@ public boolean inCompletionPhase() {
...
+     * <ul>
+     *     <li>It is a TEE task
+     *     <li>Its trust is 1
+     * </ul>
+     *
+     * @return {@literal true} if eligible, {@literal false} otherwise.
+     */
+    public boolean isEligibleToContributeAndFinalize() {
+        return isTeeTask()
+                && trust == 1;
```
Comment (jbern0rd) [inline src/main/java/com/iexec/core/task/Task.java:194]: There is a difference with iexec-commons-poco where we check if there is a callback as well.

Comment (jbern0rd) [reply src/main/java/com/iexec/core/task/Task.java:194]: Wouldn't it be better to implement the same check ?

File: src/main/java/com/iexec/core/task/Task.java [hunk 2]
```diff
@@ -178,4 +178,5 @@ public boolean inCompletionPhase() {
     public boolean isTeeTask() {
         return TeeUtils.isTeeTag(getTag());
     }
+
```
Comment (mcornaton) [inline src/main/java/com/iexec/core/task/Task.java:181]:
```suggestion
```

**Review by jbern0rd — APPROVED** [review]
LGTM :rocket:

**Pull request merged**